bot/cogs/match.py

```diff
@@ -424,18 +423,15 @@ async def _process_vote(self, reaction, user):
             if self.future is not None:
                 self.future.set_result(None)
 
-    async def vote(self):
+    async def vote(self, map_pool):
```

Comment:
Can we have `map_pool` have a default argument of `None` in which case it will set to the guild's map pool? Move the stuff on line 538 into this function. When calling for the first time we don't have to specify the map pool and then can pass it in for the recursive calls.